std-bits/table/src/main/java/org/enso/table/data/index/MultiValueIndex.java

```diff
@@ -0,0 +1,62 @@
+package org.enso.table.data.index;
+
+import org.enso.table.data.column.builder.object.*;
+import org.enso.table.data.column.builder.object.StringBuilder;
+import org.enso.table.data.column.storage.Storage;
+import org.enso.table.data.table.Column;
+import org.enso.table.data.table.Table;
+import org.enso.table.data.table.aggregate.AggregateColumnDefinition;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+public class MultiValueIndex {
+  private final Map<MultiValueKey, List<Integer>> locs;
+
+  public MultiValueIndex(Storage[] keyColumns) {
+    this.locs = new HashMap<>();
+    if (keyColumns.length != 0) {
+      int size = keyColumns[0].size();
+      for (int i = 0; i < size; i++) {
+        int finalI = i;
+        MultiValueKey key = new MultiValueKey(Arrays.stream(keyColumns).map(c -> c.getItemBoxed(finalI)).toArray());
+        List<Integer> its = this.locs.computeIfAbsent(key, x -> new ArrayList<>());
+        its.add(i);
```

Comment:
```suggestion
        List<Integer> ids = this.locs.computeIfAbsent(key, x -> new ArrayList<>());
        ids.add(i);
```

Not sure what `its` would stand for here?